pandas/core/arrays/datetimelike.py

```diff
@@ -804,6 +804,16 @@ def searchsorted(self, value, side="left", sorter=None):
         indices : array of ints
             Array of insertion points with the same shape as `value`.
         """
+        if is_list_like(value) and not isinstance(value, type(self)):
```

Comment:
The tests added for PeriodIndex were failing so had to include this bug fix

Reply:
does this need to go before the str and _recognized_scalar checks below?  IIRC putting them in this order is a pattern in this and related modules

Reply:
I don't think so, can move it down below